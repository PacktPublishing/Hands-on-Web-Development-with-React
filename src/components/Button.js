import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
  padding: 12px 48px;
  border-radius: 4px;
  border: none;
  transition: 160ms all;
  cursor: pointer;
  font-size: 16px;
  :disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }  
`;

export const PrimaryButton = Button.extend`
  background-color: ${props => props.theme.colors.primary};
  color: #fff;
  :hover {
    background-color: ${props => props.theme.colors.secondary};;
  }
`;

export const SubtleButton = Button.extend`
  color: ${props => props.theme.colors.primary};
  :hover {
    color: ${props => props.theme.colors.secondary};
  }
`;

