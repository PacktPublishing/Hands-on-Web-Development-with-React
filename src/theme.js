export default {
  colors: {
    primary: 'royalblue',
    secondary: 'crimson',
    text: 'gray',
  }
};
