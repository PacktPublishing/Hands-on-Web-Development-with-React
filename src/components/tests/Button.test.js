import React from 'react';
import renderer from 'react-test-renderer';
import { PrimaryButton, SubtleButton } from '../Button';
import theme from '../../theme';

const log = (e) => console.log(e);

describe('Button.js', () => {
  it('PrimaryButton renders correctly', () => {
    const tree = renderer
      .create(<PrimaryButton onClick={log} theme={theme}>Hello!</PrimaryButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('SubtleButton renders correctly', () => {
    const tree = renderer
      .create(<SubtleButton onClick={log} theme={theme}>SubtleButton!</SubtleButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
