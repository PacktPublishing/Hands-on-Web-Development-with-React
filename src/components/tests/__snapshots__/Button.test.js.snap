// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.js PrimaryButton renders correctly 1`] = `
<button
  className="sc-bwzfXH biIyjb"
  onClick={[Function]}
>
  Hello!
</button>
`;

exports[`Button.js SubtleButton renders correctly 1`] = `
<button
  className="sc-htpNat kKoBtB"
  onClick={[Function]}
>
  SubtleButton!
</button>
`;
